WordPress/Classes/ViewRelated/Reader/Detail/Views/ReaderDetailFeaturedImageView.swift

```diff
@@ -0,0 +1,386 @@
+import UIKit
+
+protocol ReaderDetailFeaturedImageViewDelegate: class {
+    func didTapFeaturedImage(_ sender: CachedAnimatedImageView)
+}
+
+class ReaderDetailFeaturedImageView: UIView, NibLoadable {
+    struct Constants {
+        struct multipliers {
+            static let maxPortaitHeight: CGFloat = 0.70
+            static let maxPadPortaitHeight: CGFloat = 0.50
+            static let maxLandscapeHeight: CGFloat = 0.30
+        }
+    }
+
+    struct Styles {
+        static let startTintColor: UIColor = .white
+        static let endTintColor: UIColor = UIColor(light: .black, dark: .white)
+    }
+
+    // MARK: - IBOutlets
+    @IBOutlet weak var imageView: CachedAnimatedImageView!
+    @IBOutlet weak var visualEffectView: UIVisualEffectView!
+    @IBOutlet weak var gradientView: UIView!
+    @IBOutlet weak var heightConstraint: NSLayoutConstraint!
+
+    // MARK: - Public: Properties
+    weak var delegate: ReaderDetailFeaturedImageViewDelegate?
+    var isLoading: Bool = false
+    var isLoaded: Bool = false
+
+    // MARK: - Private: Properties
+
+    /// Image loader for the featured image
+    ///
+    private lazy var imageLoader: ImageLoader = {
+        // Allow for large GIFs to animate on the detail page
+        return ImageLoader(imageView: imageView, gifStrategy: .largeGIFs)
+    }()
+
+    /// The reader post that the toolbar interacts with
+    private var post: ReaderPost?
+    private weak var scrollView: UIScrollView?
+    private weak var navigationBar: UINavigationBar?
+
+    /// An observer of the number of likes of the post
```

Comment:
Wrong comment

Reply:
Done in: a244c4c5ba05725b1b0ec935853c924e045c7673